blog-posts/takeuntil-takewhile/takeWhile-takeUntil-takeWhat.md

```diff
@@ -0,0 +1,106 @@
+
+## About TakeWhile
+
+Well, let's take a look at a quick example. The first code snippet we are looking at is using `takeWhile` to unsubscribe from an observable. I used an Angular Component here, to couple the lifecycle of the observable to the lifecycle of the component.
+Short disclaimer: It doesn't necessarily need to be an Angular component, it's just the framework I'm the most familiar with.
+
+```ts
+@Component({...})
+export class MyComponent implements OnInit, OnDestroy {
+
+  private alive = true;
+
+  public ngOnInit() {
+    myAwesomeObservable$.pipe(
+      takeWhile(val => this.alive)
+    ).subscribe();
+  }
+
+  public ngOnDestroy() {
+    this.alive = false;
+  }
+
+}
+```
+
+In this code snippet, we have a class property describing the state of the observable. It is passed to the `takeWhile` operator and as soon as this property turns false the observable should be unsubscribed from.
+This way of handling subscription is quite imperative and therefore seems to be easier for people to understand.
+
+## About TakeUntil
+
+```ts
+@Component({...})
+export class MyComponent implements OnInit, OnDestroy {
+
+  private destroyed = new Subject();
+
+  public ngOnInit() {
+    myAwesomeObservable$.pipe(
+      takeUntil(this.destroyed$)
+    ).subscribe();
+  }
+
+  public ngOnDestroy() {
+    this.destroyed.next();
+    this.destroyed.complete();
+  }
+}
+```
+
+From an high-level perspective the code snippets don't look that different. Instead of having a boolean property, describing the state of our observable, we now use a `Subject`.
+We pass the `Subject` instance to the `takeUntil` operator and as soon as we want to unsubscribe from our Observable we emit a notification over our `Subject` with the `next` method. The call of the `complete` method isn't necessary for this example, but it is a good practice within the `takeUntil`-Pattern.
+
+## The Problem
+
+So all in all both code snippets look similar and behave similar, right? Well, No!
+Let's have a look at the marble diagrams describing those operators because this will highlight the difference between those two operators.
+
+![takeUntil marble diagram, kindly provided by Michael Hladky](./assets/takeUntil.png)
+_takeUntil marble diagram, kindly provided by [Michael Hladky](https://twitter.com/Michael_Hladky)_
+
+![takeWhile marble diagram, kindly provided by Michael Hladky](./assets/takeWhile.png)
+_takeWhile marble diagram, kindly provided by [Michael Hladky](https://twitter.com/Michael_Hladky)_
+
+The problem here is that `takeWhile` is intended to take an incoming notification, use some comparison on it, which might lead to an unsubscribe. The important fact is, that `takeWhile` is triggered by the incoming notification and might unsubscribe afterwards. `TakeUntil` in contrast is triggered by the passed Observable.
```

Comment:
>use some comparison on it
I would slightly rephrase this. It takes an incoming notification, and check a specified condition. It does not necessarily need to be a comparison.